File: vignettes/01_hello_agent/csharp/01_hello_agent.cs
// Hello Agent — Simplest possible agent (C#)
//
// This sample creates a minimal agent using OllamaChatClient,
// sends a single prompt, and prints the response. It mirrors the Julia
// vignette 01_hello_agent.
//
// Prerequisites:
//   - Ollama running locally with qwen3:8b pulled
//   - dotnet restore

using Microsoft.Agents.AI;
using Microsoft.Extensions.AI;
using OllamaSharp;

var endpoint = Environment.GetEnvironmentVariable("OLLAMA_ENDPOINT")
    ?? "http://localhost:11434";
var modelName = Environment.GetEnvironmentVariable("OLLAMA_MODEL_NAME")
    ?? "qwen3:8b";

// Create an agent backed by the local Ollama instance.
AIAgent agent = new OllamaApiClient(new Uri(endpoint), modelName)
    .AsAIAgent(
        instructions: "You are a friendly assistant. Keep your answers brief.",
        name: "HelloAgent");

// Non-streaming: get the complete response at once.
Console.WriteLine(await agent.RunAsync("What is the capital of France?"));

// Streaming: receive tokens as they are generated.
Console.Write("Agent (streaming): ");
await foreach (var update in agent.RunStreamingAsync("Tell me a one-sentence fun fact."))
{
    Console.Write(update);
}
Console.WriteLine();

File: vignettes/01_hello_agent/python/01_hello_agent.py
"""
Hello Agent — Simplest possible agent (Python)

This sample creates a minimal agent using the Ollama chat client,
sends a single prompt, and prints the response. It mirrors the Julia
vignette 01_hello_agent.

Prerequisites:
  - Ollama running locally with `qwen3:8b` pulled
  - pip install agent-framework-ollama
"""

import asyncio

from agent_framework.ollama import OllamaChatClient


async def main() -> None:
    # Create a chat client pointing at the local Ollama instance.
    client = OllamaChatClient(
        host="http://localhost:11434",
        model_id="qwen3:8b",
    )

    # Build an agent with a name and system instructions.
    agent = client.as_agent(
        name="HelloAgent",
        instructions="You are a friendly assistant. Keep your answers brief.",
    )

    # Run the agent with a simple prompt (non-streaming).
    result = await agent.run("What is the capital of France?")
    print(f"Agent: {result}")

    # Streaming: receive tokens as they are generated.
    print("Agent (streaming): ", end="", flush=True)
    async for chunk in agent.run("Tell me a one-sentence fun fact.", stream=True):
        if chunk.text:
            print(chunk.text, end="", flush=True)
    print()


if __name__ == "__main__":
    asyncio.run(main())
